L1Trigger/Phase2L1ParticleFlow/src/CaloClusterer.cc

```diff
@@ -624,6 +691,108 @@ void l1tpf_calo::FlatCaloLinker::run() {
   }
 }
 
+l1tpf_calo::CombinedCaloLinker::CombinedCaloLinker(const edm::ParameterSet &pset,
+                                                   const SingleCaloClusterer &ecal,
+                                                   const SingleCaloClusterer &hcal)
+    : SimpleCaloLinkerBase(pset, ecal, hcal), combClusterer_(pset) {}
+
+l1tpf_calo::CombinedCaloLinker::~CombinedCaloLinker() {}
+
+void l1tpf_calo::CombinedCaloLinker::clear() {
+  clearBase();
+  combClusterer_.clear();
+}
+
+void l1tpf_calo::CombinedCaloLinker::run() {
+  combClusterer_.clear();
+
+  const EtGrid &hraw = hcal_.raw();
+  const EtGrid &eraw = ecal_.raw();
+  const EtaPhiCenterGrid &eeta = ecal_.etaCenter();
+  const EtaPhiCenterGrid &ephi = ecal_.phiCenter();
+  combClusterer_.raw() = eraw;
+  combClusterer_.raw() += hraw;
+
+  combClusterer_.run();
+  clusterIndex_ = combClusterer_.indexGrid();
+  const std::vector<Cluster> &clustersSrc = combClusterer_.clusters();
+  unsigned int nclust = clustersSrc.size();
+  clusters_.resize(nclust);
+  for (unsigned int ic = 0; ic < nclust; ++ic) {
+    const Cluster &src = clustersSrc[ic];
+    CombinedCluster &dst = clusters_[ic];
+    dst.et = src.et;
+    dst.eta = src.eta;
+    dst.phi = src.phi;
+    dst.ecal_et = 0;
+    dst.hcal_et = 0;
+    float pt_max = 0.;
+    float eta_ecal = 0.;
+    float phi_ecal = 0.;
+    for (const auto &pair : src.constituents) {
+      if (eraw[pair.first]) {
+        float ept = pair.second * eraw[pair.first];
+        dst.ecal_et += ept;
+        dst.constituents.emplace_back(-pair.first - 1, pair.second);
+        if (ept > pt_max) {
+          eta_ecal = eeta[pair.first];
+          phi_ecal = ephi[pair.first];
+          pt_max = ept;
+        }
+      }
+      if (hraw[pair.first]) {
+        dst.hcal_et += pair.second * hraw[pair.first];
+        dst.constituents.emplace_back(+pair.first + 1, pair.second);
+      }
+    }
+    dst.ecal_eta = eta_ecal;
+    dst.ecal_phi = phi_ecal;
+  }
+}
+
+l1tpf_calo::GridSelector::GridSelector(std::vector<double> etaBounds,
+                                       std::vector<double> phiBounds,
+                                       std::vector<unsigned int> maxClusters)
+    : etaBounds_(etaBounds),
+      phiBounds_(phiBounds),
+      maxClustersEtaPhi_(maxClusters),
+      regionPtIndices_(maxClusters.size() > 0 ? maxClusters.size() : 1) {}
```

Comment:
`scram b code-checks` wants this to be changed to `(!maxClusters.empty() ? maxClusters.size() : 1)`